src/MessagePack/Internal/UnicodeUtility.cs

```diff
@@ -0,0 +1,232 @@
+namespace MessagePack.Internal
+{
+    using System;
+    using System.Runtime.CompilerServices;
+    /// <summary>
+    /// unicode manupilation utilities
+    /// </summary>
+    public static class UnicodeUtility
+    {
+        [ThreadStatic]
+        const char UnicodeInvalidChar = (char)0xfffd;
```

Comment:
ThreadStatic should mark to under Utf8Buffer?

Reply:
I made mistake, I will fix it.